src/operations/runtime.ts

```diff
@@ -90,12 +92,79 @@ export async function getContainersProcessTrees(
       containersProcessTrees[criId] = {};
     } else {
       containersProcessTrees[criId] = data.ongoing.containers[0];
+      containersProcessTrees[criId].processes = reduceProcesses(data.ongoing.containers[0].processes, processesLimit);
     }
   }
 
   return containersProcessTrees;
 }
 
+function reduceProcesses(processes: any[], limit: number): any[] {
+  if (processes.length === 0 || limit <= 0) {
+    return [];
+  }
+
+  const countProcesses = (procs: any[]): number => {
```

Comment:
I'd suggest you pull these functions out to the global scope given they are "functional" and don't mutate anything in the function they are declared in